requirements.txt

```
@@ -1,5 +1,5 @@
 torch >= 2.2.0.dev
```

I think we need to put the `torchdata` dependency here, maybe try
```suggestion
torch >= 2.2.0.dev
--find-links https://download.pytorch.org/whl/nightly/cpu/
torchdata >= 0.7.1.dev20240426+cpu
```
If works, we can remove the dependency in unit test workflow.

---

I attempted thus but it fails to find the nightly, instead installs the latest released version. We do plan to do a torchdata release soon (which will contain the latest StatefulDataLoader) and once that happens, we can remove the explicit pip installs in unit test workflows and main README.